### PR TITLE
Clarify basic authentication

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds section about basic auth in Installation > Authentication doc
- Clarifies auth/test and /users endpoints apply to basic auth user credentials
- Adds section about `sensuctl user test-creds` command in sensuctl reference
- Clarifies basic auth info in RBAC reference

## Motivation and Context
Closes #https://github.com/sensu/sensu-docs/issues/2159

### CODE REVIEW

**Comment by hillaryfraley** [comment]
If you think this satisfies the concerns described in https://github.com/sensu/sensu-docs/issues/2159, I can propagate back t o 5.14 before merging.

File: content/sensu-go/5.17/api/auth.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ HTTP/1.1 200 OK
 
 /auth (GET)          |     |
 ---------------------|------
-description          | Generates an access token to the API using basic authentication. Access tokens last for approximately 15 minutes. When your token expires, you should see a `401 Unauthorized` response from the API. To generate a new access token, use the [`/auth/token` API endpoint](#authtoken-post).
+description          | Generates an access token and a refresh token to the API using Sensu's basic authentication. Access tokens last for approximately 15 minutes. When your token expires, you should see a `401 Unauthorized` response from the API. To generate a new access token, use the [`/auth/token` API endpoint](#authtoken-post).
```
Comment (cwjohnston) [inline content/sensu-go/5.17/api/auth.md]:
```suggestion
description          | Generates an access and a refresh token used for accessing the API using Sensu's basic authentication. Access tokens last for approximately 15 minutes. When your token expires, you should see a `401 Unauthorized` response from the API. To generate a new access token, use the [`/auth/token` API endpoint](#authtoken-post).
```

File: content/sensu-go/5.17/api/users.md [hunk 2]
```diff
@@ -26,6 +26,8 @@ menu:
   - [`/users/:user/groups/:group` (PUT)](#usersusergroupsgroup-put)
   - [`/users/:user/groups/:group` (DELETE)](#usersusergroupsgroup-delete)
 
+_**NOTE**: The users API allows you to create and manage user credentials with Sensu's built-in [basic authentication][6]. To configure user credentials with an external provider like [Lightweight Directory Access Protocol (LDAP)][4] or [Active Directory][5], use Sensu's [authentication providers API][3]._
```
Comment (cwjohnston) [inline content/sensu-go/5.17/api/users.md]:
```suggestion
_**NOTE**: The users API allows you to create and manage user credentials with Sensu's built-in [basic authentication provider][6]. To configure user credentials with an external provider like [Lightweight Directory Access Protocol (LDAP)][4] or [Active Directory (AD)][5], use Sensu's [authentication providers API][3]._
```

File: content/sensu-go/5.17/api/users.md [hunk 3]
```diff
@@ -90,7 +92,7 @@ output         | {{< highlight shell >}}
 
 ### `/users` (POST)
 
-The `/users` API endpoint provides HTTP POST access to create a [user][1].
+The `/users` API endpoint provides HTTP POST access to create a [user][1] using Sensu's basic authentication.
```
Comment (cwjohnston) [inline content/sensu-go/5.17/api/users.md]:
```suggestion
The `/users` API endpoint provides HTTP POST access to create a [user][1] using Sensu's basic authentication provider.
```

File: content/sensu-go/5.17/installation/auth.md [hunk 4]
```diff
@@ -25,15 +27,25 @@ menu:
...
 
 Sensu requires username and password authentication to access the [Sensu dashboard][1], [API][8], and command line tool ([sensuctl][2]).
-For Sensu's [default user credentials][3] and more information about configuring Sensu role-based access control (RBAC), see the [RBAC reference][4] and learn how to [create read-only users][5].
+You can use Sensu's built-in basic authentication or an external authentication provider.
 
-In addition to built-in RBAC, Sensu includes [commercial][6] support for authentication using external authentication providers.
-Sensu supports Microsoft Active Directory (AD) and standards-compliant Lightweight Directory Access Protocol (LDAP) tools like OpenLDAP.
+## Use built-in basic authentication
+
+Sensu's built-in basic authentication allows you to create and manage user credentials (usernames and passwords) with the [users API][53], either directly or using [sensuctl][2].
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
Sensu's built-in basic authentication provider allows you to create and manage user credentials (usernames and passwords) with the [users API][53], either directly or using [sensuctl][2].
```

File: content/sensu-go/5.17/installation/auth.md [hunk 5]
```diff
@@ -25,15 +27,25 @@ menu:
...
+Basic authentication does not require any external configuration.
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
The basic authentication provider does not depend on external services and is not configurable.
```

File: content/sensu-go/5.17/installation/auth.md [hunk 6]
```diff
@@ -25,15 +27,25 @@ menu:
   - [Okta](#okta)
 
 Sensu requires username and password authentication to access the [Sensu dashboard][1], [API][8], and command line tool ([sensuctl][2]).
-For Sensu's [default user credentials][3] and more information about configuring Sensu role-based access control (RBAC), see the [RBAC reference][4] and learn how to [create read-only users][5].
+You can use Sensu's built-in basic authentication or an external authentication provider.
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
You can use Sensu's built-in basic authentication provider or configure external authentication providers to authenticate via Lightweight Directory Access Protocol (LDAP), Active Directory (AD), or OpenID Connect.
```

File: content/sensu-go/5.17/api/overview.md [hunk 7]
```diff
@@ -154,7 +154,7 @@ To regenerate a valid access token, run any sensuctl command (like `sensuctl eve
 
 ### Authenticate with an API key
 
-The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username.
+Sensu's API key (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username.
```
Comment (cwjohnston) [inline content/sensu-go/5.17/api/overview.md]:
```suggestion
Each Sensu API key (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username.
```

File: content/sensu-go/5.17/reference/rbac.md [hunk 8]
```diff
@@ -262,14 +261,16 @@ You can configure `agent` user credentials with the [`user` and `password` agent
...
 
 {{< highlight shell >}}
 sensuctl user test-creds USERNAME --password 'password'
 {{< /highlight >}}
 
 An empty response indicates valid credentials.
-A request-unauthorized response indicates invalid credentials.
+A `request-unauthorized` response indicates invalid credentials.
+
+_**NOTE**: The `sensuctl user test-creds` command tests passwords for users created with Sensu's built-in [basic authentication][34]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][43] or [Active Directory][44]._
```
Comment (cwjohnston) [inline content/sensu-go/5.17/reference/rbac.md]:
```suggestion
_**NOTE**: The `sensuctl user test-creds` command tests passwords for users created with Sensu's built-in [basic authentication provider][34]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][43] or [Active Directory (AD)][44]._
```

File: content/sensu-go/5.17/api/auth.md [hunk 9]
```diff
@@ -57,7 +57,9 @@ response codes       | <ul><li>**Valid credentials**: 200 (OK)</li><li> **Invali
 
 ### `/auth/test` (GET) {#authtest-get}
 
-The `/auth/test` API endpoint provides HTTP GET access to test user credentials.
+The `/auth/test` API endpoint provides HTTP GET access to test basic authentication user credentials that were created with Sensu's built-in [basic authentication][1].
+
+_**NOTE**: The `/auth/test` endpoint only tests user credentials created with Sensu's built-in [basic authentication][1]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][2] or [Active Directory][3]._
```
Comment (cwjohnston) [inline content/sensu-go/5.17/api/auth.md]:
```suggestion
_**NOTE**: The `/auth/test` endpoint only tests user credentials created with Sensu's built-in [basic authentication provider][1]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][2] or [Active Directory (AD)][3]._
```

File: content/sensu-go/5.17/sensuctl/reference.md [hunk 10]
```diff
@@ -191,6 +192,25 @@ You can use global flags with most sensuctl commands.
...
+To test the password for a user created with Sensu's built-in [basic authentication][44]:
+
+{{< highlight shell >}}
+sensuctl user test-creds USERNAME --password 'password'
+{{< /highlight >}}
+
+An empty response indicates valid credentials.
+A `request-unauthorized` response indicates invalid credentials.
+
+_**NOTE**: The `sensuctl user test-creds` command tests passwords for users created with Sensu's built-in [basic authentication][44]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][26] or [Active Directory][42]._
```
Comment (cwjohnston) [inline content/sensu-go/5.17/sensuctl/reference.md]:
```suggestion
_**NOTE**: The `sensuctl user test-creds` command tests passwords for users created with Sensu's built-in [basic authentication provider][44]. It does not test user credentials defined via an authentication provider like [Lightweight Directory Access Protocol (LDAP)][26] or [Active Directory (AD)][42]._
```

File: content/sensu-go/5.17/installation/auth.md [hunk 11]
```diff
@@ -25,15 +27,25 @@ menu:
...
 
-In addition to built-in RBAC, Sensu includes [commercial][6] support for authentication using external authentication providers.
-Sensu supports Microsoft Active Directory (AD) and standards-compliant Lightweight Directory Access Protocol (LDAP) tools like OpenLDAP.
+## Use built-in basic authentication
+
+Sensu's built-in basic authentication allows you to create and manage user credentials (usernames and passwords) with the [users API][53], either directly or using [sensuctl][2].
+Basic authentication does not require any external configuration.
+
+With basic authentication, you can create and manage [default user credentials][3], configure [role-based access control (RBAC)][4], and [create read-only users][5].
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
After creating users via the basic authentication provider, you can manage their permissions via [role-based access control (RBAC)][4]. See [Create read-only users][5] for an example.
```

File: content/sensu-go/5.17/installation/auth.md [hunk 12]
```diff
@@ -619,7 +631,7 @@ For example:
 
 ## AD authentication
 
-Sensu offers commercial support for using Microsoft Active Directory (AD) for authentication to the Sensu dashboard, API, and sensuctl.
+Sensu offers [commercial support][6] for using Microsoft Active Directory (AD) for authentication to the Sensu dashboard, API, and sensuctl.
 The AD authentication provider is based on the [LDAP authentication provider][44].
 
 ### AD configuration examples
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
### Active Directory (AD) configuration examples {#ad-configuration-examples}
```

File: content/sensu-go/5.17/installation/auth.md [hunk 13]
```diff
@@ -619,7 +631,7 @@ For example:
 
 ## AD authentication
```
Comment (cwjohnston) [inline content/sensu-go/5.17/installation/auth.md]:
```suggestion
## Active Directory (AD) authentication {#ad-authentication}
```

optional suggestion -- not sure if this introduces too much out-of-scope change but this sort of bothered me

**Review by cwjohnston — CHANGES_REQUESTED** [review]
A few suggestions, mostly adding `provider` in a few places to help reinforce the concept. LMKWYT?

**Comment by hillaryfraley** [comment]
@cwjohnston I incorporated all of your suggestions, thank you! OK to propagate to 5.14?

**Review by asachs01 — APPROVED** [review]
👍 LGTM!

**Pull request merged**